buildroot/share/PlatformIO/scripts/common-dependencies.py

```diff
@@ -333,3 +340,4 @@ def check_configfile_locations():
 check_configfile_locations()
 apply_features_config()
 force_ignore_unused_libs()
+add_cpu_freq()
```

Comment:
This action would be better done in `common-dependencies-post.py` or even `common-cxxflags.py`.

Reply:
Please see discussion above. In short, it does not work for this usecase.

Reply:
I'm not sure whether we can still build STM32 in Arduino IDE. But we do want to preserve that ability without any major differences in the output.

Reply:
Let me correct the previous sentence. I've not tried moving the lines in `common-cxxflags.py`, but `common-dependencies-post.py` is not used for creating external libraries compilation's flags.

Reply:
I'll try to move into `common-cxxflags.py` and let you know.